### PR TITLE
Export fix

### PR DESCRIPTION
**Pull request by Norby09**
Worked on export and proper display of elements on text inputs

### CODE REVIEW

File: src/data/item.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ export class Item {
   backgroundColor: String;
   backgroundUrl: String;
   name: String;
-  type: number = 1;
+  type = 1;
```
Comment (danvatamaniuc) [inline src/data/item.ts]: I'd say keep the type here, and add it everywhere else. Makes life easier :D

File: src/app/item/item.component.ts [hunk 2]
```diff
@@ -19,11 +19,15 @@ export class ItemComponent implements OnInit {
...
+      const element = this.items[this.items.length - 1];
+      itemId = element.id;
+      this.items.push(new Item({id: ++itemId}));
+    }
+  }
 
-  removeItem(item : Item) {
-    if(this.items.length > 1) {
+  removeItem(item: Item) {
+    if (this.items.length > 1) {
```
[the remark this answers is not in the document]
Comment (Norby09) [reply src/app/item/item.component.ts]: yes

File: src/app/itemList/itemList.component.ts [hunk 3]
```diff
@@ -36,10 +38,40 @@ export class ItemListComponent implements OnInit{
     this.type = Type.create();
     this.setting = Settings.create();
   }
-
   export(format: string = 'json'): string {
+  debugger;
```
[the remark this answers is not in the document]
Comment (Norby09) [reply src/app/itemList/itemList.component.ts]: deleted them

File: src/app/itemList/itemList.component.ts [hunk 4]
```diff
@@ -36,10 +38,40 @@ export class ItemListComponent implements OnInit{
...
+    console.log(this.languages);
```
[the remark this answers is not in the document]
Comment (Norby09) [reply src/app/itemList/itemList.component.ts]: deleted

File: src/app/itemList/itemList.component.ts [hunk 5]
```diff
@@ -36,10 +38,40 @@ export class ItemListComponent implements OnInit{
...
+        // resource key names for existing i18n strings
+        , beforeLangKeys = beforeLang && beforeLang.resources.map(element => element.name)
+      ;
 
-    this.languages = this.languageService.getLanguagesAndResources();
+      if (!beforeLang) {
+        this.languages.push(currentLang);
+      }
+      for (let j = 0 ; j < currentLangKeys.length ; j++ ) {
+        if (beforeLang !== undefined) {
```
[the remark this answers is not in the document]
Comment (Norby09) [reply src/app/itemList/itemList.component.ts]: done

File: src/app/itemList/itemList.component.ts [hunk 6]
```diff
@@ -90,18 +122,29 @@ export class ItemListComponent implements OnInit{
         this.languageObj = json.i18n;
 
         this.languages = [];
-
+        console.log(this.languageObj);
```
[the remark this answers is not in the document]
Comment (Norby09) [reply src/app/itemList/itemList.component.ts]: deleted

File: src/app/languages.service.ts [hunk 7]
```diff
@@ -46,4 +51,30 @@ export class LanguagesService {
...
+  // this should return an object like this:
+  // getResourceValue('@title1');
+  // { "en": "This is my title", "ru": "сука блядь"}
+  // getResourceValue('@title1', 'en');
+  // "This is my title";
+
+  getResourceValue(resourceKey: string, lang?: string): string {
+    const resourceValue = {};
+    let langObj = '';
+    console.log(this.i18n);
```
[the remark this answers is not in the document]
Comment (Norby09) [reply src/app/languages.service.ts]: done

File: src/app/languages.service.ts [hunk 8]
```diff
@@ -46,4 +51,30 @@ export class LanguagesService {
...
+    if (!lang && this.i18n) {
+      this.i18n.map(languageObj => {
+        const resource = languageObj.resources.find(res => res.name === resourceKey);
+        langObj = languageObj.name;
+        resource && (resourceValue[languageObj.name] = resource.value);
+        console.log(resourceValue);
```
[the remark this answers is not in the document]
Comment (Norby09) [reply src/app/languages.service.ts]: done

File: src/app/languages.service.ts [hunk 9]
```diff
@@ -46,4 +51,30 @@ export class LanguagesService {
...
+  // { "en": "This is my title", "ru": "сука блядь"}
+  // getResourceValue('@title1', 'en');
+  // "This is my title";
+
+  getResourceValue(resourceKey: string, lang?: string): string {
+    const resourceValue = {};
+    let langObj = '';
+    console.log(this.i18n);
+    if (!lang && this.i18n) {
+      this.i18n.map(languageObj => {
```
[the remark this answers is not in the document]
Comment (Norby09) [reply src/app/languages.service.ts]: changed

File: src/app/types/types.component.html [hunk 10]
```diff
@@ -1,22 +1,28 @@
...
-        </label>
-    </div>
-    <div class="col-lg-4">
-        <label class="col-lg-12 overwritePadding"> Custom Template : 
-            <input type="text" class="form-control" [(ngModel)]="type.customTemplate"/>
-        </label>
-    </div>
-  </div>
-</div>
+<!--<div class="bl&#45;&#45;form-layout">-->
```
[the remark this answers is not in the document]
Comment (Norby09) [reply src/app/types/types.component.html]: yes

**Pull request merged**